spring-cloud-task-batch/src/main/java/org/springframework/cloud/task/batch/listener/support/JdbcTaskBatchDao.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.task.batch.listener.support;
+
+import javax.sql.DataSource;
+
+import org.springframework.batch.core.JobExecution;
+import org.springframework.cloud.task.batch.listener.TaskBatchDao;
+import org.springframework.cloud.task.repository.TaskExecution;
+import org.springframework.cloud.task.repository.dao.JdbcTaskExecutionDao;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * JDBC based implementation of the {@link TaskBatchDao}.  Intended to be used in
+ * conjunction with the JDBC based
+ * {@link org.springframework.cloud.task.repository.TaskRepository}
+ *
+ * @author Michael Minella
+ */
+public class JdbcTaskBatchDao implements TaskBatchDao {
+
+	private String tablePrefix = JdbcTaskExecutionDao.DEFAULT_TABLE_PREFIX;
+
+	private static final String INSERT_STATEMENT = "INSERT INTO %PREFIX%TASK_BATCH VALUES(?, ?)";
+
+	private JdbcOperations jdbcTemplate;
+
+	/**
+	 * @param dataSource dataSource where the task batch table resides.
```

Comment:
The second one really should be `DataSource`...the first one is the name of the param...